### ISSUE TITLE
QField crashes when select read only layers

### ISSUE THREAD

**Issue by giedrioks26** [opengisch/QField#384]
When you select the editable layer after selecting a read only layer and then back to the editable layer, QFied crashes.
Same with non-editable layer of sqlite views

It starts with version 0.11.x. No problem with 0.10.13.
I add a sample project: point and line layers is read-only. The polygon is an editable layer.
[read_only](https://user-images.githubusercontent.com/6436791/47273428-d29e3c00-d59c-11e8-9924-9f50c42568be.png)

sample project: 
[null.zip](https://github.com/opengisch/QField/files/2499694/null.zip)

Android 8.0, device: Blackview BV 9000 pro

How it works:
![bug](https://user-images.githubusercontent.com/6436791/47273441-f82b4580-d59c-11e8-8787-27bb36e42d70.gif)

**Comment by giedrioks26** [comment]
The problem persists on new QField version 0.11.10

**Comment by giedrioks26** [comment]
The problem stays with 1.0.0 rc3.
1. crashes when qfield opens new and has not yet entered data. 
2. When I enter the first data (point, line, polygon) then qfield does not crashes in this session.

**Comment by 3nids** [comment]
I could reproduce the bug on the phone but not on computer

backtrace:
```
01-31 14:11:32.125 17323 17365 W QField  : qrc:/qml/DashBoard.qml:202:5: QML Legend: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
01-31 14:11:32.384  1447  1615 E UsbDeviceManager: handle message = 9
01-31 14:11:32.385  2255  2255 D KeyguardUpdateMonitor: received broadcast android.intent.action.BATTERY_CHANGED
01-31 14:11:32.388  1447  3949 V LocSvc_HIDL_Subscription_jni: battery_level_update
01-31 14:11:32.390  1447  3949 V LocSvc_HIDL_Subscription_jni: [battery_level_update][931] [HC] =>> [HS]
01-31 14:11:32.391   813   813 V LocSvc_HIDL_IzatSubscription: [batteryLevelUpdate][777] [HS] <<<<= [HC]
01-31 14:11:32.391  2255  2255 D KeyguardUpdateMonitor: handleBatteryUpdate
01-31 14:11:32.392  1447  3949 V LocSvc_HIDL_Subscription_jni: Exit Result 0
01-31 14:11:32.759  2472  2496 I Finsky  : [223] com.google.android.finsky.bo.an.run(6): Stats for Executor: Db-scheduler com.google.android.finsky.bo.ao@f7326a5[Running, pool size = 1, active threads = 0, queued tasks = 0, completed tasks = 124]
01-31 14:11:32.760  2472  2496 I Finsky  : [223] com.google.android.finsky.bo.an.run(6): Stats for Executor: Db-scheduler_logging_store.db com.google.android.finsky.bo.ao@26bea7a[Running, pool size = 1, active threads = 0, queued tasks = 0, completed tasks = 20]
01-31 14:11:32.761  2472  2496 I Finsky  : [223] com.google.android.finsky.bo.an.run(6): Stats for Executor: InstallQueueDatabaseManager com.google.android.finsky.bo.ao@dc0322b[Running, pool size = 1, active threads = 0, queued tasks = 0, completed tasks = 136]
01-31 14:11:33.009 17323 17365 W QField  : qrc:/qml/FeatureForm.qml:141:11: QML Rectangle: Cannot anchor to an item that isn't a parent or sibling.
01-31 14:11:33.010 17323 17365 W QField  : qrc:/qml/FeatureForm.qml:178:26: Unable to assign [undefined] to QModelIndex
01-31 14:11:33.568 17323 17365 W QField  : <Unknown File>: QML ObjectModel: remove: indices [-1 - 0] out of range [0 - 1]
01-31 14:11:34.068 17323 17365 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x20 in tid 17365 (qtMainLoopThrea), pid 17323 (.opengis.qfield)
01-31 14:11:34.214 17402 17402 I crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
01-31 14:11:34.215  1000  1000 I /system/bin/tombstoned: received crash request for pid 17365
01-31 14:11:34.217 17402 17402 I crash_dump32: performing dump of process 17323 (target tid = 17365)
01-31 14:11:34.252 17402 17402 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-31 14:11:34.252 17402 17402 F DEBUG   : Build fingerprint: 'xiaomi/jasmine/jasmine_sprout:9/PKQ1.180904.001/V10.0.2.0.PDIMIFJ:user/release-keys'
01-31 14:11:34.252 17402 17402 F DEBUG   : Revision: '0'
01-31 14:11:34.252 17402 17402 F DEBUG   : ABI: 'arm'
01-31 14:11:34.252 17402 17402 F DEBUG   : pid: 17323, tid: 17365, name: qtMainLoopThrea  >>> ch.opengis.qfield <<<
01-31 14:11:34.252 17402 17402 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x20
01-31 14:11:34.252 17402 17402 F DEBUG   : Cause: null pointer dereference
01-31 14:11:34.252 17402 17402 F DEBUG   :     r0  d1629e40  r1  00050b03  r2  840e8dfe  r3  00000000
01-31 14:11:34.252 17402 17402 F DEBUG   :     r4  d1629e40  r5  d1627f60  r6  d17d72c0  r7  d346d3a4
01-31 14:11:34.252 17402 17402 F DEBUG   :     r8  00000000  r9  d34574c0  r10 00000001  r11 d17d7cd8
01-31 14:11:34.252 17402 17402 F DEBUG   :     ip  f4bdcd00  sp  d9af8e18  lr  d95ab0ff  pc  d95a67a8
01-31 14:11:34.253 17402 17402 F DEBUG   : 
01-31 14:11:34.253 17402 17402 F DEBUG   : backtrace:
01-31 14:11:34.253 17402 17402 F DEBUG   :     #00 pc 001367a8  /data/app/ch.opengis.qfield-dudn6KkO_Gsqc4BVJ5f5gQ==/lib/arm/libQt5Core.so (QObject::thread() const+2)
01-31 14:11:34.463   804   804 W memtrack@1.0-se: type=1400 audit(0.0:11413): avc: denied { read } for name="mem" dev="debugfs" ino=485341 scontext=u:r:hal_memtrack_default:s0 tcontext=u:object_r:qti_debugfs:s0 tclass=file permissive=0
01-31 14:11:34.973  2255  2255 V NavigationBar: onRotationProposal proposedRotation=ROTATION_0, winRotation=ROTATION_0, isValid=true, mNavBarWindowState=WINDOW_STATE_SHOWING, rotateSuggestionsDisabled=false, isRotateButtonVisible=false
01-31 14:11:35.060  1000  1000 E /system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_00
01-31 14:11:35.065  1447 17406 W ActivityManager:   Force finishing activity ch.opengis.qfield/org.qtproject.qt5.android.bindings.QtActivity
01-31 14:11:35.085  1447 17406 D ActivityTrigger: ActivityTrigger activityPauseTrigger 
01-31 14:11:35.097  1447  1599 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
```

**Comment by giedrioks26** [comment]
Sometimes it happens not with read only layers. I guess some different geometry (from point to polygon and back).

**Comment by 3nids** [comment]
not sure, I guess it's rather some issue with indexing in a model.
@m-kuhn any idea how to get more info here?
 I cannot reproduce on the computer (Qt 5.12) but I get it on the phone (Qt 5.11).

**Comment by m-kuhn** [comment]
There's a lot that changed between 0.10.13 and 0.11.10 (or did it appear on earlier versions already @giedrioks26 ?) https://github.com/opengisch/QField/compare/release-0.10.13...release-0.11.10

@3nids printf debugging is the best I can come up with so far. Maybe checking if it's related to the featureform (by replacing it with an empty rectangle) because of the warning messages before. Next thing would be to check if it's related to geometry painting (the thread() call in the trace and opengl code which runs on its own thread could be an issue). Anyway, your guess is as good as mine.

**Comment by giedrioks26** [comment]
@m-kuhn It starts with version 0.11.x. No problem with 0.10.13 and previous versions.

**Comment by m-kuhn** [comment]
@giedrioks26 do you have an idea about the minimum version of the x in 0.11.x? was it with 0.11.0 already?

**Comment by giedrioks26** [comment]
@m-kuhn yes, problem start from 0.11.0 (including 0.11.0). In previous versions work fine.

**Comment by m-kuhn** [comment]
Updated code comparison link
https://github.com/opengisch/QField/compare/release-0.10.13...release-0.11.0

**Comment by giedrioks26** [comment]
solved with #498
